### DISCUSSION TITLE
[Feature Request] +/- 1min intervals

### DISCUSSION THREAD

**Discussion by jaymarvels** [advplyr/audiobookshelf-app#93] [Ideas]
Similar to how we can fastfoward/rewind by 10 seconds, another option next to them to ff/rw by 1 min intervals would be handy.

I'm more of a BE dev but do have some limited vuejs knowledge and looking at the code there is already a method to skip by 10 secs that could be reused for this.

Something like this (but obviously not as ugly/fitting in the current style):

![image](https://user-images.githubusercontent.com/7057631/151856496-e143bf2e-75c2-4fb3-9072-bdc7379d9177.png)

**Comment by jcrmit** [comment]
@jaymarvels - thanks. +1 this feature. Tghe typical use case for me is going to sleep and either forgeting the sleep timer or fallen alseep within 1 min! :) - Ability to skip back 1min would be something I wold use daily. Otherwise - i find my self jumping chapters or scrubing the seek bars.

**Reply by jaymarvels** [reply]
yeah similar, I usually have a 15min sleep timer on but often fall asleep before it ends so shifting back in 1min intervals could make it easier than rapidly tapping 10 secs

**Comment by advplyr** [comment]
I just worry about cluttering the UI. What if the skip backwards interval was 30s instead of 10s?
The android notification controls only allow for 1 forward & 1 backward button, same with android auto.
An option could be added to configure that skip interval.

**Reply by jaymarvels** [reply]
I personally like the idea of both, but understand the possible need of a compromise to fit on various device widths etc.

I have a S21+ and a quick mockup looks a bit like this

![image](https://user-images.githubusercontent.com/7057631/152001975-4fede3b1-92a8-4f42-81f4-18b93e24ce8a.png)

**Reply by jcrmit** [reply]
I think a 30 sec button would be a reasonable compromise if you only wanted one button (user configurable would be great). 

I don't think the UI cluttering should be a major concern, but accept it need to be as  consistent as possible across platforms and interfaces.

Thanks for considering 👍

**Reply by scottdk** [reply]
I raised a feature request https://github.com/advplyr/audiobookshelf-app/issues/695 over in Issues without realising this discussion was here.

My angle on it is that I have ADHD, and quite often zone out and miss a chunk of minutes of a book and want to quickly jump back several minutes to find where I was. Sometimes I'll go back too far trying to find where I was, so then I want to quickly jump forward by minutes rather than 10 seconds. 

But when my inattentiveness is short I might only miss a single critical word, so I jump back 10 seconds.

Using the location slider to adjust is troublesome when the chapters are long. Thankfully you ingeniously added the history item. 

In mAbook, the previous app I used before discovering this one, I was using both +/-1 min and +/-10 seconds **_very_** often. (I swear it takes me 20% longer to get through books because I repeat so much 😅) 

![Screenshot_20230506_120435_mAbook.jpg](https://user-images.githubusercontent.com/11282782/236592770-ab8b0a36-a6ca-4c0a-8a42-2556c32e0585.jpg)

**Reply by dedors** [reply]
I would love this feature too! 
SAP goes far and even offers 5 different button layouts.

**Reply by samszotkowski** [reply]
![image](https://github.com/advplyr/audiobookshelf-app/assets/94253934/26be7967-f07f-4a5a-b75a-18a442d36b0e)

Here's a mockup I made (0.9.65 android), I think moving the chapter skip buttons up by where the chapter title is displayed would be intuitive and leave the UI feeling uncluttered.  This should alleviate any concerns about device width, because the chapter skip buttons are relocated and we're occupying up their previous space.

As far as the Settings UI, the forward/back skips could be condensed into one option for "small skip duration" used for both forward and back, and add another for "large skip duration."  Android notification, android auto, etc. could use the small skip duration and blissfully ignore the other.

30s skip is not a very good compromise in my opinion, for the two situations of "I missed a word" and "I just zoned out for 5 minutes" anything over like 15s feels annoying for the former.

**Reply by jaymarvels** [reply]
The other option is to add in more options in settings, rather than 5, 10, 30 you could add in one more "60"

![Screenshot_20230829_190808_audiobookshelf](https://github.com/advplyr/audiobookshelf-app/assets/7057631/5e6ab2b0-1782-4d71-bfc9-96182ba6dfad)

**Reply by samszotkowski** [reply]
That does solve the "zoned out for 5 minutes" problem, but it un-solves the "missed a word" problem (any duration over 15s I'd say this is true).  I think two buttons would be ideal for people that encounter both those problems, or I guess it doesn't necessarily have to be buttons if my mockup looks too cluttered for example, maybe something like a long-press or double-tap to do a big skip.  I'd argue dedicated buttons are the most intuitive though.

I should also note I made #849 before seeing this.

**Comment by wesf99** [comment]
> ![image](https://private-user-images.githubusercontent.com/94253934/264105320-26be7967-f07f-4a5a-b75a-18a442d36b0e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUyNzY5MTEsIm5iZiI6MTcwNTI3NjYxMSwicGF0aCI6Ii85NDI1MzkzNC8yNjQxMDUzMjAtMjZiZTc5NjctZjA3Zi00YTVhLWI3NWEtMThhNDQyZDM2YjBlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE0VDIzNTY1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmZWRhODFjN2U2ZmI3NjVkMTliNDYwNDgxZjkwZDA1MTgxYjljMGI3NGE4NWJhNDcyNDU4ODEyNTQ0YjEyMmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.BXlonWOGyQHMrNmVJRP5SnoL4YtRaZJwRkF_gTxQQxM)
> 
> Here's a mockup I made (0.9.65 android), I think moving the chapter skip buttons up by where the chapter title is displayed would be intuitive and leave the UI feeling uncluttered. This should alleviate any concerns about device width, because the chapter skip buttons are relocated and we're occupying up their previous space.
> 
> As far as the Settings UI, the forward/back skips could be condensed into one option for "small skip duration" used for both forward and back, and add another for "large skip duration." Android notification, android auto, etc. could use the small skip duration and blissfully ignore the other.
> 
> 30s skip is not a very good compromise in my opinion, for the two situations of "I missed a word" and "I just zoned out for 5 minutes" anything over like 15s feels annoying for the former.

I think this would be great feature. Uncluttered and functional. Allow customizable long and short skip forward/backward durations in settings.

**Reply by mimicscott** [reply]
+1.
> > FWIW, [Smart Audiobook Player] (https://play.google.com/store/apps/details?id=ak.alizandro.smartaudiobookplayer) implements controls like this and I think they work quite well. That said, it feels unlikely that this will help, given that even if both apps use the same frameworks, ABS also has to try to match ios's capabilities as well. HTH but if not, carry on!
